group-generator.js

```diff
@@ -18,53 +18,49 @@ let period1 = [
   {name: 'Gollum', level: 2}
 ];
 
-
 function addRandomSortNumber(array) {
-  array.map(function(student){
-  student.sortNumber = Math.random();
-
+  array.map(function (student) {
+    student.sortNumber = Math.random();
   });
   return array;
 };
 
 function addLeveledSortNumber(array) {
-  array.map(function(student) {
+  array.map(function (student) {
     student.sortNumber = student.level + Math.random();
-  })
+  });
   return array;
 }
 
 function addOrderedSortNumber(array) {
   let i = 0;
-  array.map(function(student) {
+  array.map(function (student) {
     student.sortNumber = i;
-    i ++;
-  })
+    i++;
+  });
   return array;
 }
 
-function sort(array){
-  array.sort(function(a, b) {
-  return a.sortNumber - b.sortNumber;
-});
+function sort(array) {
+  array.sort(function (a, b) {
+    return a.sortNumber - b.sortNumber;
+  });
+
   return array;
 };
 
 function arrayOfNames(array) {
-  let nameArray =[]
-  array.forEach(function(student) {
+  let nameArray = [];
+  array.forEach(function (student) {
     nameArray.push(student.name);
-  })
+  });
   return nameArray;
 };
 
-
-
 function group(array, size) {
   let newArray = [];
-  let actual
   newArray.push(array.slice(0, size));
-  for(i = size; i < array.length; i = i + size) {
+  for (let i = size, len = array.length; i < len; i = i + size) {
```

Comment:
This is the only actual performance recommendation. The first line of a `for` loop is evaluated only once. The second and third lines, however, are reevaluated each time you loop. E.g. `Is 1 < 4? Yes.` `Is 2 < 4? Yes.` etc. When you keep the `array.length` in the second part of the for loop, you are reevaluating that array length each and every time. It's not a noticeable performance issue, but when you have a HUGE js application, those things can add up.

Recommend pre-calculating the length of the array, so that `array.length` value is stored in memory. One common way to do that is to throw the length into a file beforehand:
```js
let arrLength = array.length;
for (let i = 0; i < arrLength; i++) {...}
```
I prefer to double up my vars in the for loop, since the second argument only happens after a semicolon. Thus:
```js
for (let i = size, len = array.length; i < len; i = i + size) {...}
```

Also, your `i` variable was undeclared. So, I added the `let` to the for loop.